go/pkg/services/logcollector/server.go

```diff
@@ -331,6 +340,33 @@ func (s *Server) GetLogs(request *protologcollector.GetLogsRequest, responseStre
 	return nil
 }
 
+// HasLogs returns true if the log file exists for a given run id
+func (s *Server) HasLogs(ctx context.Context, request *protologcollector.HasLogsRequest) (*protologcollector.HasLogsResponse, error) {
+
+	s.Logger.DebugWithCtx(ctx, "Received Has Logs request", "runUID", request.RunUID)
+
+	// get log file path
+	_, err := s.getLogFilePath(ctx, request.RunUID, request.ProjectName)
+	if err != nil {
+		if strings.Contains(common.GetErrorStack(err, 10), "not found") {
```

Comment:
why not checking the `errors.RootCause(err).Is(os.FileNotFoundErr)` or something similar

Reply:
It is a custom search for the file, and hence a custom error.
I don't want to return an `os` error, but I can create a custom `NotFound` error maybe.
I agree about the `RootCasue` tho.